kernel/mmanutils.c

```diff
@@ -984,57 +984,46 @@ int myst_msync(void* addr, size_t length, int flags)
 /* notified on close to clear fd-mappings involving fd */
 void myst_mman_close_notify(int fd)
 {
-    int flags;
     struct stat buf;
     bool locked = false;
 
-    /* get the file open flags */
-    if (fd < 0 || (flags = myst_syscall_fcntl(fd, F_GETFL, 0)) < 0)
```

Comment:
Do we need to retain the fd < 0 check?

Reply:
The sole user of this function `myst_syscall_close`, does check for a valid fd. But added the check back to be defensive.